lib/src/main/java/io/ably/lib/transport/Hosts.java

```diff
@@ -19,16 +21,23 @@ private Hosts() { /* Restrict new instance creation */ }
 	 * @return Successor host that can be used as a fallback.
 	 * null, if there is no successor fallback available.
 	 */
-	public static String getFallback(String host) {
-		int size = FALLBACKS.size();
-		int indexCurrent = FALLBACKS.indexOf(host);
+	public static String getFallback(String host, List<String> fallbackHosts) {
+
+		if(fallbackHosts == null)
+			return null;
+
+		int size = fallbackHosts.size();
```

Comment:
I don't see where the hosts are randomised, `RSC15a` stats `try fallback hosts in random order`.  In other libraries, at the time the instance is created, the hosts are randomised once and persisted in random order.  If that is done, then all other logic will continue to work as is.